lfs/pointer_smudge.go

```diff
@@ -154,8 +154,76 @@ func readLocalFile(writer io.Writer, ptr *Pointer, mediafile string, cb CopyCall
 		}
 	}
 
+	if len(ptr.Extensions) > 0 {
+		registeredExts := Config.Extensions()
+		extensions := make(map[string]Extension)
+		for _, ptrExt := range ptr.Extensions {
+			ext, ok := registeredExts[ptrExt.Name]
+			if !ok {
+				err := fmt.Errorf("Extension '%s' is not configured.", ptrExt.Name)
+				return Error(err)
+			}
+			ext.Priority = ptrExt.Priority
+			extensions[ext.Name] = ext
+		}
+		exts, err := SortExtensions(extensions)
+		if err != nil {
+			return Error(err)
+		}
+
+		// pipe extensions in reverse order
+		var extsR []Extension
+		for i, _ := range exts {
+			ext := exts[len(exts)-1-i]
+			extsR = append(extsR, ext)
+		}
+
+		request := &pipeRequest{"smudge", reader, workingfile, extsR}
+
+		var response pipeResponse
+		response, err = pipeExtensions(request)
```

Comment:
err was already declared, but I suppose there's not much harm in declaring it again.